include/zigbee/zigbee_app_utils.rst

```diff
@@ -277,29 +282,49 @@ It is used in :c:func:`zigbee_default_signal_handler` by default.
 
 If the network is left by a router or an end device, the device tries to join any open network.
 
-* The router uses the default signal handler to try to join or rejoin the network until it succeeds.
-* The end device uses the default signal handler to try to join or rejoin the network for a finite period of time, because the end devices are often powered by batteries.
+The Zigbee rejoin procedure retries to join a network with each try after ``2^n`` seconds, where ``n`` is the number of retries.
+The period is limited to the time specified in ``REJOIN_INTERVAL_MAX_S``, which by default equals 15 minutes.
 
-  * The procedure to join or rejoin the network is restarted after the device reset or power cycle.
-  * The procedure to join or rejoin the network can be restarted by calling :c:func:`user_input_indicate`, but it needs to be implemented in the application (for example, by calling :c:func:`user_input_indicate` when a button is pressed).
-    The procedure is restarted only if the device does not join and the procedure is not running.
+Router rejoining network
+++++++++++++++++++++++++
 
-The Zigbee rejoin procedure retries to join a network with each try after a specified amount of time: ``2^n`` seconds, where ``n`` is the number of retries.
+The router uses the default signal handler to try to join or rejoin the network until it succeeds.
 
-The period is limited to 15 minutes if the result is higher than that.
+When :c:func:`start_network_rejoin` is called, the rejoin procedure is started, and depending on the device role:
 
-* When :c:func:`start_network_rejoin` is called, the rejoin procedure is started, and depending on the device role:
+When ``stop_network_rejoin(was_scheduled)`` is called, the network rejoin is canceled and the alarms scheduled by :c:func:`start_network_rejoin` are canceled.
 
-  * For the end device, the application alarm is scheduled with ``stop_network_rejoin(ZB_TRUE)``, to be called after the amount of time specified in ``ZB_DEV_REJOIN_TIMEOUT_MS``.
-    Once called, the alarm stops the rejoin.
+  * Additionally for the end device, if :c:func:`stop_network_rejoin` is called with ``was_scheduled`` set to ``ZB_TRUE``, :c:func:`user_input_indicate` can restart the rejoin procedure.
```

Comment:
This sentence is in the `Router rejoining network` section. If it is applicable for the router devices - I think the condition (`for the end device`) should be removed. If it is not - why it was not moved completely to the `End device rejoining network` section?